Refactors deathsounds, adds 2 new ones (lizards and borgs)

PKPenguin321 opened the pull request:
🆑
code: Xeno deathsounds are less snowflakey. Deathsounds are now easy to add to mobs in code.
soundadd: Lizards and borgs now have sounds that play on *deathgasp, like xenos. So as to not nerf silent kills, anything that mutes (gags, poison, being a mime) or having 50 or more oxygen damage will prevent you from making a sound when you deathgasp.
/:cl:

- deathsounds are sounds played when *deathgasp is used, either manually or automatically
- deathsounds are a var on /mob
- made xenos use the var
- removed the hack for xenos deathsound in the deathgasp emote and made it use the var
- 2 new deathsounds: one for lizards, one for borgs

**ALL SOUNDS ARE SUBJECT TO CHANGE.** Please give feedback or suggestions if you have an improvement or better sounds. Info on the lizard sound can be found in the credits.txt I added, and the borg deathsound is a completely unused one from old TF2 builds that I can't find a license to (I'm not sure it even has a license on it). I'd like to maybe add human deathsounds too, but I can't find any sounds I like enough (and I would want to add male/female variants).

[Here is a video with sound to demonstrate the new sounds.](https://file.house/tBRs.mp4) Outdated now, see the new borg deathsound linked in comments below.

- [x] Lizard deathsound isn't applied to people who pick lizard race roundstart - needs to be fixed before merge
- [x] Make the sound not play if you're mute (mimes or poison or otherwise), maybe make it not play if you have too much oxy damage so you can do silent chokeouts/leave people in crit for them to die silently

PKPenguin321 committed:
refactors deathsounds, adds 2 new ones
- deathsounds are sounds played when *deathgasp is used, either manually or automatically
- deathsounds are a var on /mob
- made xenos use the var
- removed the hack for xenos deathsound in the deathgasp emote and made it use the var
- 2 new deathsounds: one for lizards, one for borgs

Poojawa commented on code/modules/mob/living/carbon/human/human.dm:
```diff
@@ -997,6 +997,7 @@
 
 /mob/living/carbon/human/species/lizard
 	race = /datum/species/lizard
+	deathsound = 'sound/voice/lizard/deathsound.ogg'
```
Make this sort of thing an On species gain/lose sort of thing. That'll sort out all of the sounds

PKPenguin321 replied:
I ended up figuring this out and fixing it up on my own before seeing this, but thank you for your help regardless!

Poojawa replied:
Yee yee. Does it still apply the correct sound if you swap species?

PKPenguin321 committed:
fixes and balance
- alt species now get their deathsound properly and have it set in their species datum
- people who are silent (poisoned, mimes, or otherwise) wont make a deathsound
- people who have over 50 oxyloss when they die wont make a deathsound (to allow silent kills)

ShizCalev commented on code/modules/mob/living/carbon/human/species.dm:
```diff
@@ -44,6 +44,7 @@ GLOBAL_LIST_EMPTY(roundstart_races)
 	var/siemens_coeff = 1 //base electrocution coefficient
 	var/damage_overlay_type = "human" //what kind of damage overlays (if any) appear on our species when wounded?
 	var/fixed_mut_color = "" //to use MUTCOLOR with a fixed color that's independent of dna.feature["mcolor"]
+	var/deathsound//used to set the mobs deathsound on species change
```
put a space before the // please.

ShizCalev commented on code/modules/mob/living/emote.dm:
```diff
@@ -87,8 +87,12 @@
+			var/mob/living/L = user
+			if(!L.can_speak_vocal() || L.oxyloss >= 50)
+				return//stop the message if: oxyloss too high/cant speak
+		playsound(user.loc, user.deathsound, 80, 1, 1)
```
drop the .loc

ATH1909 commented:
Give bluspace golems the enderman death noise/sound as their death sound.

anconfuzedrock commented:
Moth deathscream?!?!?!

Dorsisdwarf commented:
borg *deathgasp sound should probbably be more in line with their standard harsh electric *buzz *beep *buzz2 etc

PKPenguin321 commented:
@Dorsisdwarf i liked your idea, tried my hand at it. let me know what you think
**[new borg deathsound link](https://file.house/AGb-.mp3)**

PKPenguin321 committed:
addresses review
makes some comments prettier/better, drops a .loc

actioninja commented:
I could record myself screaming like I'm dying if you want.

PKPenguin321 commented:
If you'd like to and think you can make a sound that's clean enough to put in the game and that players will like, go ahead

anconfuzedrock commented:
the new sound is loud, it should be like when they start up, but "ow"

AnturK reviewed (dismissed):
simple animals will now have both `deathsound` and `death_sound`. Unify these.

PKPenguin321 commented:
Good point, will do

PKPenguin321 commented:
@aconfuzedrock
Which sound are you referring to?

PKPenguin321 commented:
@ anconfuzedrock
Which sound are you referring to?

anconfuzedrock commented:
I honestly have no idea pk is a joke

improvedname commented:
Now time for human deathsound

improvedname commented:
Its only fair why should lizard/borgs get death sound alarms to nearby people when the glorious master race humans. Cant have roblox oof

PKPenguin321 committed:
new borg deathsound
with credits.txt

PKPenguin321 committed:
moves simple animal death_sound to new deathsound
code for simple animal deathsound has been changed around in simple_animal.dm to tie it to deathgasp like everything else

PKPenguin321 commented on code/modules/mob/living/simple_animal/hostile/megafauna/drake.dm:
```diff
@@ -60,11 +60,11 @@ Difficulty: Medium
+	deathsound = 'sound/magic/demon_dies.ogg'
 	var/datum/action/small_sprite/smallsprite = new/datum/action/small_sprite/drake()
-	
+
```
accidentally deleted a bunch of whitespace in this file, can undo if maintainers want
